src/ECEC-Model-Tests/ECECForagerTest.class.st

```diff
@@ -69,13 +69,37 @@ ECECForagerTest >> testHarvestRate [
 ]
 
 { #category : 'tests' }
-ECECForagerTest >> testInitialEnergy [
+ECECForagerTest >> testInitialCatabolicRate [
+
+	self assert: ECECForager catabolicRate equals: 2.
+]
+
+{ #category : 'tests' }
+ECECForagerTest >> testInitialEnergy [	
+		
+	self assert: ECECForager initialEnergy equals: 50.
+]
+
+{ #category : 'tests' }
+ECECForagerTest >> testInitialEnergySet [
 
 	ECECForager initialEnergy: 13.	
 		
 	self assert: ECECForager initialEnergy equals: 13.
 ]
 
+{ #category : 'tests' }
+ECECForagerTest >> testInitialFertilityThreshold [
+
+	self assert: ECECRestrainedForager fertilityThreshold equals: 100.
+]
+
+{ #category : 'tests' }
+ECECForagerTest >> testInitialHarvestRate [	
+		
+	self assert: ECECForager harvestRate equals: nil.
```

Comment:
You should test the harvest rate of restrained and unrestrained foragers, not the abstract method of a superclass